pkg/apis/cluster/v1alpha1/machinedeployment_types.go

```diff
@@ -212,26 +212,40 @@ func ValidateMachineDeploymentStrategy(strategy *cluster.MachineDeploymentStrate
 	allErrs := field.ErrorList{}
 	switch strategy.Type {
 	case common.RollingUpdateMachineDeploymentStrategyType:
-		allErrs = append(allErrs, ValidateMachineRollingUpdateDeployment(strategy.RollingUpdate, fldPath.Child("rollingUpdate"))...)
+		if strategy.RollingUpdate != nil {
+			allErrs = append(allErrs, ValidateMachineRollingUpdateDeployment(strategy.RollingUpdate, fldPath.Child("rollingUpdate"))...)
+		}
+	default:
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("Type"), strategy.Type, "is an invalid type"))
 	}
 	return allErrs
 }
 
 func ValidateMachineRollingUpdateDeployment(rollingUpdate *cluster.MachineRollingUpdateDeployment, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, ValidatePositiveIntOrPercent(rollingUpdate.MaxUnavailable, fldPath.Child("maxUnavailable"))...)
-	allErrs = append(allErrs, ValidatePositiveIntOrPercent(rollingUpdate.MaxSurge, fldPath.Child("maxSurge"))...)
-	maxUnavailable, _ := getIntOrPercent(rollingUpdate.MaxUnavailable, false)
-	maxSurge, _ := getIntOrPercent(rollingUpdate.MaxSurge, true)
-	if maxUnavailable == 0 && maxSurge == 0 {
+	var maxUnavailable int
+	var maxSurge int
+
+	if rollingUpdate.MaxUnavailable != nil {
+		allErrs = append(allErrs, ValidatePositiveIntOrPercent(rollingUpdate.MaxUnavailable, fldPath.Child("maxUnavailable"))...)
+		maxUnavailable, _ = getIntOrPercent(rollingUpdate.MaxUnavailable, false)
+
+		// Validate that MaxUnavailable is not more than 100%.
+		if len(utilvalidation.IsValidPercent(rollingUpdate.MaxUnavailable.StrVal)) == 0 && maxUnavailable > 100 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("maxUnavailable"), rollingUpdate.MaxUnavailable, "should not be more than 100%"))
```

Comment:
I'm pretty sure you'll need to escape the `%` here. Can you test?

Reply:
I reran the tests and it passes. the automated system also verified it builds/tests.
As it is standard string, and not a formatted string, I don't think escaping is needed here.